src/staking/block_validator.cpp

```diff
@@ -258,6 +319,50 @@ class BlockValidatorImpl : public AbstractBlockValidator {
     return CheckCoinbaseTransactionInternal(block, coinbase_tx, nullptr, nullptr);
   }
 
+  BlockValidationResult CheckTransaction(
```

Comment:
Not sure if our current tests are covering this logic atm.

Reply:
Some of them are covered. I will add more tests.

This is a stand-in for `CheckTransaction` for `tx_verify` which was needed in order to make use of parameters from blockchain parameters and return block validation result errors. The old function did not have unit tests. This stuff is checked through functional tests, but I agree it should be unit tested, I will add.

Reply:
Added tests in https://github.com/dtr-org/unit-e/pull/1106/commits/6cdc2df173dacfdac906954b98b834f19c866dd9